### PR TITLE
Add support for extra host

### PR DESCRIPTION
**Pull request by tdomzal**
Allow adding of entries to containers /etc/hosts with
host config 'extra hosts' option. It is equivalent
of 'docker run --add-host .. '.

Signed-off-by: Tomasz Domzal <tdomzal@gmail.com>

### CODE REVIEW

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 1]
```diff
@@ -1332,43 +1332,79 @@ public void testPauseContainer() throws Exception {
...
+    assertThat(actual.extraHosts(), equalTo(expected.extraHosts()));
 
     final String logs;
-    try (LogStream stream = sut.logs(info.id(), stdout(), stderr())) {
-      logs = stream.readFully();
+    try (LogStream stream = sut.logs(id, stdout(), stderr())) {
+        logs = stream.readFully();
     }
-    assertThat(logs, containsString("bar"));
+    assertThat(logs, containsString("1.2.3.4"));
```
Comment (mattnworb) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: is this just testing that the logs contain the IP address `1.2.3.4` from adding it to `/etc/hosts`? I would worry that something might change in the future with regard to logging that string where the actual functionality still works but the test fails.

**Comment by mattnworb** [comment]
Besides comment above looks good to me :+1:

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java (hunk 1 above)
Comment (tdomzal) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Not logs in general, but stdout from cmd to be exact.
My intention was checking contents of /etc/hosts in container so figured simple method would be just 'cat' it's contents and grep to leave extrahost line only.
So test executes single piped command via shell: 

    cat /etc/hosts | grep extrahost

and expects single line with `1.2.3.4` in stdout.

Maybe those small corrections would make it more explicit and rebust?
* change 'sut.logs(...)' to read stdout only (to exclude possibility that stderr  contains the same IP pattern)
* rename variable 'logs' to 'stdoutLogs' to express it's purpose - checking stdout only
* make sure stdoutLogs will contain single line matching string with `1.2.3.4` and ending with `extrahost`

What do you think ?

Comment (mattnworb) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Looks like I misinterpreted this part of the test - `cat /etc/hosts | grep extrahost` and checking the output sounds robust to me.

I misinterpreted this as streaming the logs of the original container starting up

Comment (tdomzal) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Great. 
Thanks !

**Pull request merged**